crates/sui-sdk/src/transaction_builder.rs

```diff
@@ -214,10 +214,16 @@ impl TransactionBuilder {
         let obj_ref = obj.compute_object_reference();
         let owner = obj.owner;
         objects.insert(id, obj);
-        Ok(if owner.is_shared() {
-            ObjectArg::SharedObject(id)
-        } else {
-            ObjectArg::ImmOrOwnedObject(obj_ref)
+        Ok(match owner {
+            Owner::Shared {
+                initial_shared_version,
+            } => ObjectArg::SharedObject {
+                id,
+                initial_shared_version,
+            },
+            Owner::AddressOwner(_) | Owner::ObjectOwner(_) | Owner::Immutable => {
```

Comment:
@lxfind / @mystenmark / @tnowacki -- from the Rust SDK perspective, is this the support we're talking about, to map from `ObjectID` to an `ObjectArg` paying attention to the `initial_shared_version`?